From 711178aa171ea467e3ab41489fcb11b24b6984df Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Thu, 19 Jun 2014 14:13:21 -0400
Subject: [cpufreq] intel_pstate: Add CPU IDs for Broadwell processors

Message-id: <20140619141321.29352.23385.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 84274
O-Subject: [PATCH RHEL7.1 BZ1086874] intel_pstate: Add CPU IDs for Broadwell processors
Bugzilla: 1086874
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1086874

Description:
------------

Add support for Broadwell processors.

Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

RHEL Version Found:
-------------------
RHEL 7.0

Brew:
-----
https://brewweb.devel.redhat.com/taskinfo?taskID=7592130

Upstream:
---------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=c7e241df5970171e3e86a516f91ca8a30ca516e8

Test Status:
------------
Tested by me using intel-sharkbay-ult-02.

[    0.000000] Detected CPU family 6 model 61

Once the Broadwell ids are added to the table of supported processors you will see similar msgs as below.

[    2.409484] Intel pstate controlling: cpu 0
[    2.414251] Intel pstate controlling: cpu 1
[    2.419009] Intel pstate controlling: cpu 2
[    2.423758] Intel pstate controlling: cpu 3

---------------------------------------------------------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index d51f17e..6b2e1ef 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -590,10 +590,13 @@ static const struct x86_cpu_id intel_pstate_cpu_ids[] = {
  ICPU(0x37, byt_params),
  ICPU(0x3a, core_params),
  ICPU(0x3c, core_params),
+ ICPU(0x3d, core_params),
  ICPU(0x3e, core_params),
  ICPU(0x3f, core_params),
  ICPU(0x45, core_params),
  ICPU(0x46, core_params),
+ ICPU(0x4f, core_params),
+ ICPU(0x56, core_params),
  {}
 };
 MODULE_DEVICE_TABLE(x86cpu, intel_pstate_cpu_ids);
-- 
1.7.1